airflow/www/views.py

```diff
@@ -1718,6 +1730,7 @@ def run(self, session=None):
         dag_run_id = request.form.get('dag_run_id')
         map_index = request.args.get('map_index', -1, type=int)
         origin = get_safe_url(request.form.get('origin'))
+        return_as_json = request.form.get('return_as_json') == "true"
```

Comment:
I wonder if it’s better to use HTTP header for this. We could use `request.content_type` (automatically return JSON for `application/json` and redirect otherwise), or the conventional `X-Requested-With`.

Reply:
Good call. I added a check for `request.headers['Accept'] == 'application/json'` instead of a param